scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -111,6 +116,34 @@ class FossId(
 
             return projectName
         }
+
+        /**
+         * This function fetches credentials for [repoUrl] and insert them between the url scheme and the host. If no
+         * matching host is found the in netrc file, the [repoUrl] is returned untouched.
+         */
+        fun getCredentialsFromNetrc(repoUrl: String): String {
+            val matcher = HOST_EXTRACTOR_REGEX.matchEntire(repoUrl)
+            return if (matcher?.groupValues?.size == 4) {
+                val (_, scheme, host, remainder) = matcher.groupValues
+
+                log.info { "Checking .netrc file for host $host ..." }
+
+                val creds = OrtAuthenticator(Authenticator.getDefault()).requestPasswordAuthenticationInstance(
```

Comment:
Rather than manually creating an `OrtAuthenticator` every time, I think this function should just rely on the default authenticator (and assume that an `OrtAuthenticator` has been installed properly).

This would also help with the testing problems you mentioned: In the test, you could temporarily replace the default `Authenticator` with a mock one and thus get configurable credentials. (Just remember to restore the original default `Authenticator` afterwards, maybe inside a *try*-*finally* block or by means of the KoTest framework.

Reply:
Well spotted.

In the end I can use `getDefault()` because the `OkHttpClientHelper` installs it.